src/models/Location.java

```diff
@@ -1,31 +1,54 @@
 package models;
 
 import logic.State;
+import logic.TransitionSystem;
+import logic.Pruning;
 
 import java.util.*;
-import java.util.stream.Collectors;
 
-public class Location {
-    protected String name;
-    protected int x, y;
-
-    protected Guard invariantGuard;
-    protected CDD invariantCdd;
-
-    protected CDD inconsistentPart;
-
-    protected boolean isInitial;
-    protected boolean isUrgent;
-    protected boolean isUniversal;
-    protected boolean isInconsistent;
-
-    protected List<Location> productOf;
-    protected Location location;
+/**
+ * {@link Location} is a class used by both {@link Automaton} and {@link TransitionSystem} to decribe a locaton.
+ * It is named and has coordinates describing the position it should be drawn.
+ * A {@link Location} can be marked as initial, urgent, universal, and inconsistent.
+ * In order to reduce the conversions between {@link Guard} and {@link CDD} the invariant is stored as both and only updated when required.
+ * For {@link Pruning} it also stores the inconsistent part of its invariant.
+ * <p>
+ * A {@link Location} can also consist of multiple locations, if this is the case then it is a product.
+ * This is the case when combining multiple locations e.g. when performing {@link logic.Conjunction}.
+ * For the product it is assumed that the order of location also corresponds to the child {@link TransitionSystem}.
+ * <p>
+ * A {@link Location} can also be a simple location which is nearly the same as a product with one element.
```

Comment:
You mention that a simple location is almost the same as a product with one element. Can you describe what it exactly is, so future developers don't have to guess that?

Reply:
The line below describes why it is "nearly" the same: "However, if the location is simple then the lazy evaluation of its invariant will always be the invariant of the child.". Maybe "The difference between a product with one element is that the simple location will always return the invariant of its child." highlights it better?

Reply:
I have changed the javadoc for the class, and I personally think it is more descriptive, but the definition is still vague.